src/components/JobCard/JobCard.tsx

```diff
@@ -1,68 +1,77 @@
 import { Text, View, Pressable, Modal } from 'react-native';
 import styles from './CardStyles';
 import React, { useState } from 'react';
+import GestureRecognizer from 'react-native-swipe-gestures';
 
 interface JobCardProps {
-  title: string;
   description: string;
-  employer: string;
   hours: number;
-  salary: number;
+  employer: string;
   contact_info: string;
-  end_date: Date;
-  job_creator: string;
-  start_date: string;
+  id: string;
+  date: string;
+  companyName: string;
+  address: string;
+  contactPerson: string;
+  phone: string;
+  jobPosition: string;
+  languageRequirement: string;
+  workingHours: string;
+  workingDays: string;
+  salary: string;
+  probationPeriod: string;
+  employeeBenefit: string;
+  otherInfo: string;
+  hidden: Map<string, boolean>;
 }
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type, @typescript-eslint/naming-convention
-const JobCard = ({ title, description, employer, hours, salary, contact_info }: JobCardProps) => {
+const JobCard = ({ title, description, employer, hours, salary, contact_info, hidden }: JobCardProps) => {
   const [modalVisible, setModalVisible] = useState(false);
 
   return (
-    <View style={styles.cardContainer}>
-      <Modal
-        transparent={true}
-        visible={modalVisible}
-        animationType="slide"
-        onRequestClose={() => {
+    <Pressable
+      style={styles.cardContainer}
+      onPress={() => {
+        setModalVisible(true);
+      }}>
+      <GestureRecognizer
+        style={{ flex: 1 }}
+        onSwipeDown={() => {
           setModalVisible(!modalVisible);
         }}>
-        <View style={styles.centeredView}>
-          <View style={styles.modalView}>
-            <View style={styles.modalHeader}>
-              <Text style={styles.titleText}>{title}</Text>
-            </View>
-            <View style={styles.modalInfo}>
+        <Modal
+          transparent={true}
+          visible={modalVisible}
+          animationType="slide"
+          onRequestClose={() => {
+            setModalVisible(!modalVisible);
+          }}>
+          <View style={styles.centeredView}>
+            <View style={styles.modalView}>
+              <View style={styles.modalHeader}>
+                <Text style={styles.modalJobRefText}>{title}</Text>
+                <Text style={styles.modalJobNameText}>{title}</Text>
+              </View>
+              <View style={styles.modalInfo}>
               <Text style={styles.modalText}>employer: {employer}</Text>
-              <Text style={styles.modalText}>hours: {hours}</Text>
-              <Text style={styles.modalText}>salary: {salary}</Text>
-              <Text style={styles.modalText}>{description}</Text>
-              <Text style={styles.modalText}>contact: {contact_info}</Text>
+               <Text style={styles.modalText}>hours: {hours}</Text>
+                <Text style={styles.modalText}>salary: {salary}</Text>
```

Comment:
You should go through the firebase and make sure that all of the jobs follow the same schema as the up to date version of jobs that Charles merged into main so we don't get any inconsistencies with the displayed job cards — I'll add this as a sprint task for you to have visibility.